src/lib/mina.ts

```diff
@@ -307,6 +316,29 @@ function LocalBlockchain({
     sendTransaction(txn: Transaction) {
       txn.sign();
 
+      let commitments = Ledger.transactionCommitments(
+        JSON.stringify(zkappCommandToJson(txn.transaction))
+      );
+
+      txn.transaction.accountUpdates.forEach(async (party) => {
```

Comment:
We should await these promises at the level of `sendTransaction`. Right now, they're escaping the synchronous context this runs in. This requires refactoring of `sendTransaction` to be async -- which is a breaking change in the API, but fine IMO, "sending" is something inherently async, this won't complicate any existing flows of using snarkyjs